app/src/main/java/valentyn/androidtasks/datadetails/cities/CityActivity.kt

```diff
@@ -0,0 +1,183 @@
+package valentyn.androidtasks.datadetails.cities
+
+import android.app.Activity
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.view.Menu
+import android.view.MenuItem
+import valentyn.androidtasks.R
+import com.squareup.picasso.Picasso
+import kotlinx.android.synthetic.main.activity_city.*
+import valentyn.androidtasks.datadetails.DataDetailPresenter
+import android.content.Intent
+import android.graphics.Color
+import android.net.Uri
+import android.provider.MediaStore
+import android.support.design.widget.TextInputLayout
+import valentyn.androidtasks.datadetails.DataDetailContract
+import valentyn.androidtasks.utils.ErrorTypeTextValidate
+import valentyn.androidtasks.utils.FileUtils
+
+class CityActivity : AppCompatActivity(), DataDetailContract.View {
+
+    private var presenter: DataDetailPresenter =
+        DataDetailPresenter()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_city)
+
+        setSupportActionBar(toolbar_city)
+        supportActionBar?.apply {
+            setDisplayShowTitleEnabled(false)
+            setDisplayHomeAsUpEnabled(true)
+        }
+        presenter.onAttach(this, intent.getStringExtra(CITY_KEY), CITY_KEY)
+
+        city_title_edit.addTextChangedListener(presenter.getTextValidator(city_title_input))
+        city_site_edit.addTextChangedListener(presenter.getTextValidator(city_site_input))
+        city_description_edit.addTextChangedListener(presenter.getTextValidator(city_description_input))
+        city_country_edit.addTextChangedListener(presenter.getTextValidator(city_country_input))
```

Comment:
Create a method setUpTextChangeListeners() {} and work with listeners there.
Try do not create long methods